lib/std/tempfiles.nim

```diff
@@ -93,17 +94,21 @@ proc createTempFile*(prefix, suffix: string, dir = ""): tuple[fd: File, path: st
   ## If failing to create a temporary file, `IOError` will be raised.
   ##
   ## .. note:: It is the caller's responsibility to remove the file when no longer needed.
-  ##
   var dir = dir
   if dir.len == 0:
     dir = getTempDir()
 
   createDir(dir)
 
+  # xxx we need a `proc toMode(a: FilePermission): Mode`, possibly by
+  # exposing fusion/filepermissions.fromFilePermissions to stdlib; then we need
+  # to add a `perm` param so users can customize this (e.g. the temp file may
+  # need execute permissions).
+  let mode = Mode(S_IRUSR or S_IWUSR)
```

Comment:
move mode to the internal of `safeOpen` or use `when not defined(windows)`\
https://github.com/nim-lang/Nim/pull/17889/checks?check_run_id=2464253495

see CI failure

```
2021-04-29T07:01:41.8938561Z  (4, 5): 'pairs' [iterator declared in D:\a\Nim\Nim\lib\system\iterators.nim(166, 10)]
2021-04-29T07:01:41.8939384Z  (4, 5): 'pairs' [iterator declared in D:\a\Nim\Nim\lib\system\iterators.nim(185, 10)]
2021-04-29T07:01:41.8940212Z  (4, 5): 'pairs' [iterator declared in D:\a\Nim\Nim\lib\system\iterators.nim(204, 10)]
2021-04-29T07:01:41.8941013Z  (4, 5): 'repr' [proc declared in D:\a\Nim\Nim\lib\system.nim(1382, 8)]
2021-04-29T07:01:41.8941758Z  (4, 5): 'true' [enumField declared in D:\a\Nim\Nim\lib\system\basic_types.nim(16, 16)]
2021-04-29T07:01:41.8942637Z D:\a\Nim\Nim\lib\std\tempfiles.nim(107, 30) Error: undeclared identifier: 'S_IWUSR'
2021-04-29T07:01:41.8943509Z candidates (edit distance, scope distance); see '--spellSuggest': 
2021-04-29T07:01:41.8944343Z  (1, 4): 'S_IRUSR' [unknown declared in D:\a\Nim\Nim\lib\std\tempfiles.nim(107, 19)]
```

Reply:
done; in future work we need to think how to expose `mode` (and `tempPathLength`) in a cross platform way